shared/storage/simple/simple.go

```diff
@@ -22,7 +25,12 @@ type Store struct {
 	mu sync.Mutex
 	g  singleflight.Group
 
-	pool     *certpool.CertPool
+	logger slog.Logger
+
+	roots   certpool.CertPool
+	inter   certpool.CertPool
+	bundler certpool.Bundler
```

Comment:
Why switched from a pointer?

Reply:
It used to be created by a PoolBuffer, but it's refactored to allow adding certificates later instead of only at creation. This also includes the change about automatic initialisation in order to work directly on the zero-value instead of relying on a New()